torchtune/modules/transformer.py

```diff
@@ -245,3 +246,121 @@ def forward(
         # shape: [b, s, out_dim] - out_dim is usually the vocab size
         output = self.output(h).float()
         return output
+
+
+class TiedEmbeddingTransformerDecoder(TransformerDecoder):
```

Comment:
This PR is awesome!

Small comment here: similar to how `GemmaTransformerDecoder` is a separate component to `TransformerDecoder`, is it worth removing the inheritance here too and just inheriting from `nn.Module`? It should be fine to replicate the code in the constructor and `setup_caches`.

Reply:
+ 1 to this!

Reply:
> This PR is awesome!
> 
> Small comment here: similar to how `GemmaTransformerDecoder` is a separate component to `TransformerDecoder`, is it worth removing the inheritance here too and just inheriting from `nn.Module`? It should be fine to replicate the code in the constructor and `setup_caches`.

OK, I will remove the inheritance.